mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala

```diff
@@ -113,7 +113,7 @@ class BinaryClassificationMetrics @Since("3.0.0") (
       } else {
         iter
       }
-    }
+    }.cache()
```

Comment:
you can cache the returned rdd on your own purpose. 
We should not cache it here